unified-codes/apps/web/src/sagas/DetailSaga.ts

```diff
@@ -0,0 +1,112 @@
+import { call, put, takeEvery, all, delay } from 'redux-saga/effects';
+
+import {
+  AlertSeverity,
+  IEntity,
+  IAlert,
+} from '@unified-codes/data';
+
+import {
+  DETAIL_ACTIONS,
+  DetailActions,
+  AlertActions,
+  IDetailFetchEntityAction,
+} from '../actions';
+
+
+const ALERT_SEVERITY = {
+  FETCH: AlertSeverity.info,
+  ERROR: AlertSeverity.error,
+};
+
+const ALERT_TEXT = {
+  FETCH: 'Fetching...',
+  ERROR: 'Could not fetch entity.',
+};
+
+const alertError: IAlert = {
+  isVisible: true,
+  severity: ALERT_SEVERITY.ERROR,
+  text: ALERT_TEXT.ERROR,
+};
+
+const alertFetch: IAlert = {
+  isVisible: true,
+  severity: ALERT_SEVERITY.FETCH,
+  text: ALERT_TEXT.FETCH,
+};
+
+const getEntityQuery = (code: string) => `
+{
+  entity (code: "${code}") {
+    code
+    description
+    type
+    has_property {
+      type
+      value
+    }
+    # form_category
+    has_child {
+      code
+      description
+      type
+      # form
+      has_child {
+        code
+        description
+        type
+        # unit_of_use
+        has_child {
+          code
+          description
+          type
+        }
+      }
+    }
+  }
+}`;
+
+const getEntity = async (
+  url: string,
+  code: string
+): Promise<IEntity> => {
+  const response = await fetch(url, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      Accept: 'application/json',
+    },
+    body: JSON.stringify({
+      query: getEntityQuery(code),
+    }),
+  });
+  const json = await response.json();
+  const { data } = json;
+  const { entity } = data;
+  return entity;
+};
+
+function* fetchDetails(action: IDetailFetchEntityAction) {
+  yield put(AlertActions.raiseAlert(alertFetch));
+  try {
+    const url = `${process.env.NX_DATA_SERVICE_URL}:${process.env.NX_DATA_SERVICE_PORT}/${process.env.NX_DATA_SERVICE_GRAPHQL}`;
+    const entity: IEntity = yield call(getEntity, url, action.code);
```

Comment:
```suggestion
    const { code } = action;
    const entity: IEntity = yield call(getEntity, url, code);
```

Reply:
Ah yes, bit tidier :) thanks I will use the suggestion